kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkTask.java

```diff
@@ -287,14 +287,14 @@ public void flush(Map<TopicPartition, OffsetAndMetadata> offsets) {
 
   @Override
   public void close(Collection<TopicPartition> partitions) {
-    for (TopicPartition tp : topicPartitionWriters.keySet()) {
+    for (TopicPartition tp : partitions) {
       try {
         topicPartitionWriters.get(tp).close();
```

Comment:
should we check that 'tp' is indeed in topicPartitionWriters in case close is called more than once?

Reply:
That's a reasonable request - updated.